src/planet-a/Active/SysPlanetA.cpp

```diff
@@ -255,6 +217,41 @@ void SysPlanetA::update_geometry(ActiveScene& rScene)
                 return EChunkUpdateAction::Chunk;
             });
 
+            // TEMP: until we have a proper planet shader
+            static auto planetDrawFnc =
+                [](ActiveEnt e, ActiveScene& rScene, Magnum::GL::Mesh& rMesh,
+                    ACompCamera const& camera, ACompTransform const& transform)
+            {
+                using namespace Magnum;
+                auto& glResources = rScene.get_context_resources();
+                osp::DependRes<Shaders::MeshVisualizer3D> shader =
+                    glResources.get<Shaders::MeshVisualizer3D>("mesh_vis_shader");
+
+                auto& planet = rScene.reg_get<ACompPlanet>(e);
+
+                Matrix4 entRelative = camera.m_inverse * transform.m_transformWorld;
+
+                (*shader)
+                    .setColor(0x2f83cc_rgbf)
+                    .setWireframeColor(0xdcdcdc_rgbf)
+                    .setViewportSize(Vector2{Magnum::GL::defaultFramebuffer.viewport().size()})
+                    .setTransformationMatrix(entRelative)
+                    .setNormalMatrix(entRelative.normalMatrix())
+                    .setProjectionMatrix(camera.m_projection)
+                    .draw(*planet.m_mesh);
+            };
+
+            osp::Package& glResources = rScene.get_context_resources();
+
+            // Generate mesh resource
+            std::string name = osp::string_concat("planet_mesh_",
+                std::to_string(static_cast<int>(ent)));
+            planet.m_mesh = glResources.add<Magnum::GL::Mesh>(name);
```

Comment:
I know this is temporary, but this should really not be the way meshes are added normally. I mentioned this in many places already, but as-generic-as-possible components are the way to go.

PlanetGeometryA stores its own buffers, and outputs which parts of it need to be updated, not relying on the GPU at all. It's may benefit from a generalized updatable mesh for a function in the render order to handle.

Reply:
I don't really understand the planet geometry enough to quite grasp what you're saying but I think it's similar to what I'm doing with the mapscreen data. To be honest I forget how the planet mesh worked before, was it just created as a class member instead of created in the graphics resources?